ArduPlane/servos.cpp

```diff
@@ -464,6 +464,15 @@ void Plane::set_servos_controlled(void)
         // ask quadplane code for forward throttle
         SRV_Channels::set_output_scaled(SRV_Channel::k_throttle, quadplane.forward_throttle_pct());
     }
+    // For ArduSoar
+    
+    if ((control_mode == FLY_BY_WIRE_B ||
+        control_mode == AUTO || control_mode == LOITER) &&
```

Comment:
or if is_active() is the wrong spot, have your update() set active false when in the wrong modes. Point is, Soaring should care about the mode, not the plane code.